macro/src/expand.rs

```diff
@@ -2211,7 +2114,15 @@ fn expand_extern_type(ty: &Type, types: &Types, proper: bool) -> TokenStream {
         Type::RustOption(ty) => {
             let span = ty.name.span();
             let langle = ty.langle;
-            let elem = expand_extern_type(&ty.inner, types, proper);
+            let elem = match &ty.inner {
+                Type::RustBox(boxed) => {
+                    let langle = boxed.langle;
+                    let rangle = boxed.rangle;
+                    let elem = expand_extern_type(&boxed.inner, types, proper);
+                    quote_spanned!(span=> ::cxx::alloc::boxed::Box #langle #elem #rangle)
```

Comment:
this feels a bit lame/brittle though I don't think you can break it. This function blindly converts Box<T> to *T/*c_void so I can't just recurse. Maybe there is an impl somewhere else I can use though